src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -186,7 +186,8 @@ private BoundExpression BindSwitchExpression(SwitchExpressionSyntax node, Bindin
             bool permitDesignations,
             ref bool hasErrors,
             bool misplaced,
-            BindingDiagnosticBag diagnostics)
+            BindingDiagnosticBag diagnostics,
+            bool reportUnsupported)
```

Comment:
This is needed to report unsupported type for slice if we don't have a length property.
In that case we already reported unsupported type for list so we could probably live with that. However, we will have to adjust the tests to account for this which wouldn't be exactly per spec.